Mutagen.Bethesda.Analyzers/AnalyzerRunnerBuilder.cs

```diff
@@ -1,90 +1,210 @@
 ﻿using System.IO.Abstractions;
+using Autofac;
+using Mutagen.Bethesda.Analyzers.Autofac;
 using Mutagen.Bethesda.Analyzers.Config.Topic;
 using Mutagen.Bethesda.Analyzers.SDK.Topics;
+using Mutagen.Bethesda.Analyzers.Skyrim;
 using Mutagen.Bethesda.Environments;
+using Mutagen.Bethesda.Environments.DI;
+using Mutagen.Bethesda.Plugins;
 using Mutagen.Bethesda.Plugins.Cache;
 using Mutagen.Bethesda.Plugins.Order;
 using Mutagen.Bethesda.Plugins.Records;
+using Noggog;
 using Noggog.WorkEngine;
 
 namespace Mutagen.Bethesda.Analyzers;
 
-public class AnalyzerRunnerBuilder
+public record AnalyzerRunnerBuilderNeedsTarget
+{
+    private readonly GameRelease _gameRelease;
+
+    public AnalyzerRunnerBuilderNeedsTarget(GameRelease gameRelease)
+    {
+        _gameRelease = gameRelease;
+    }
+
+    public AnalyzerRunnerBuilder WithLinkCache(ILinkCache linkCache)
+    {
+        return new AnalyzerRunnerBuilder(
+            _gameRelease,
+            linkCache);
+    }
+
+    public AnalyzerRunnerBuilder WithLoadOrder(ILoadOrder<IModListingGetter<IModGetter>> loadOrder)
```

Comment:
I think this might also need a variant where you're able to pass in ILoadOrder<IModListingGetter<TModGetter>> with a generic TModGetter as this is not possibel atm.

```cs
var gameEnvironment = GameEnvironment.Typical.Skyrim(SkyrimRelease.SkyrimSE);

var analyzer = AnalyzerRunnerBuilder.Create(GameRelease.SkyrimSE)
    .WithLoadOrder(gameEnvironment.LoadOrder) // error
```